src/DynamoCore/Search/Legacy/SearchElements/SearchElementBase.cs

```diff
@@ -137,50 +125,34 @@ public class LibraryItem
         public IEnumerable<PortInfo> ReturnKeys { get; private set; }
 
 
-        public LibraryItem(SearchElementBase node, DynamoModel dynamoModel)
+        public LibraryItem(NodeSearchElement node, DynamoModel dynamoModel)
         {
             Category = node.FullCategoryName;
-            Type = node.Type;
             DisplayName = Name = node.Name;
             CreationName = node.CreationName;
             Description = node.Description;
-            Searchable = node.Searchable;
             Weight = node.Weight;
-            Keywords = dynamoModel.SearchModel.SearchDictionary.GetTags(node);
+            Keywords = dynamoModel.SearchModel.GetTags(node);
 
             PopulateKeysAndParameters(dynamoModel);
         }
 
         private void PopulateKeysAndParameters(DynamoModel dynamoModel)
         {
-            var controller = dynamoModel.EngineController;
-            var functionItem = (controller.GetFunctionDescriptor(CreationName));
+            var services = dynamoModel.LibraryServices;
+            var functionItem = services.GetFunctionDescriptor(CreationName);
             NodeModel newElement = null;
             if (functionItem != null)
             {
                 DisplayName = functionItem.DisplayName;
-                if (functionItem.IsVarArg)
-                    newElement = new DSVarArgFunction(dynamoModel.CurrentWorkspace, functionItem);
-                else
-                    newElement = new DSFunction(dynamoModel.CurrentWorkspace, functionItem);
+                newElement = (functionItem.IsVarArg)
+                    ? new DSVarArgFunction(functionItem) as NodeModel
+                    : new DSFunction(functionItem);
```

Comment:
Hi @Benglin, I remember when library items sending was implemented first time there was some issue here that `FunctionDescriptor` contains not all inputs and outputs for some nodes. I didn't go into particulars why it worked this way and just implemented the obtaining port information from `NodeModel` for `FunctionDescriptor` case as well